scripts/migration/migrate_backrefs_between_nodes_and_nodelogs.py

```diff
@@ -0,0 +1,73 @@
+"""
+This migration will add original_node and node associated with the log to nodelogs. It will then make
+copies of each nodelog for the remaining nodes in the backref (registrations and forks),
+changing the node to the current node.
+"""
+
+import sys
+import logging
+from modularodm import Q
+from website.app import init_app
+from website import models
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+logger = logging.getLogger(__name__)
+
+
+def lookup_node(node_id):
+    """
+    Retrieves original node on nodelog
+    """
+    return models.Node.find(Q('_id', 'eq', node_id))[0]
+
+
+def main(dry=True):
+    init_app(routes=False)
+    node_logs = models.NodeLog.find(Q('original_node', 'eq', None))
+    total_log_count = node_logs.count()
+    count = 0
+    errored_logs = []
+    logger.info('Adding original_node and node fields to each log and then cloning the log for remaining nodes in the log backref.')
+
+    for log in node_logs:
+        # Two steps. Step 1: Add original_node and node field to log.
+        count += 1
+        original_node = lookup_node(log.params['node'])
+        log.original_node = original_node
+        log.node = lookup_node(log._backrefs['logged']['node']['logs'][0])
```

Comment:
The script failed here with some old staging data --  the 1829th log had no backrefs. Further investigation on that outdated dump revealed that there were 46 of ~60000 broken in that way, and that they were all comment related. 

Michael checked prod for me:

```python
In [1]: models.NodeLog.find(Q('__backrefs', 'eq', None)).count()
Out[1]: 406
In [2]: bnls = models.NodeLog.find(Q('__backrefs', 'eq', None))

In [3]: set([x.action for x in bnls])
Out[3]:
{u'addon_added',
 u'comment_added',
 u'comment_removed',
 u'comment_updated',
 u'contributor_added',
 u'contributor_removed',
 u'edit_title',
 u'embargo_completed',
 u'file_added',
 u'file_removed',
 u'github_file_added',
 u'made_private',
 u'node_forked',
 u'osf_storage_file_added',
 u'wiki_updated'}
```